udp_mux_multi.go

```diff
@@ -95,20 +194,29 @@ func NewMultiUDPMuxFromPort(port int, opts ...UDPMuxFromPortOption) (*MultiUDPMu
 		return nil, err
 	}
 
-	conns := make([]net.PacketConn, 0, len(ips))
+	conns := make([]net.PacketConn, 0, len(ports)*len(ips))
 	for _, ip := range ips {
-		conn, listenErr := params.net.ListenUDP("udp", &net.UDPAddr{IP: ip, Port: port})
-		if listenErr != nil {
-			err = listenErr
-			break
-		}
-		if params.readBufferSize > 0 {
-			_ = conn.SetReadBuffer(params.readBufferSize)
+		for _, port := range ports {
+			conn, listenErr := params.net.ListenUDP("udp", &net.UDPAddr{IP: ip, Port: port})
+			if listenErr != nil {
+				err = listenErr
+				break
+			}
+			if params.readBufferSize > 0 {
+				_ = conn.SetReadBuffer(params.readBufferSize)
+			}
+			if params.writeBufferSize > 0 {
+				_ = conn.SetWriteBuffer(params.writeBufferSize)
+			}
+			if params.batchWriteSize > 0 {
+				conns = append(conns, tudp.NewBatchConn(conn, params.batchWriteSize, params.batchWriteInterval))
```

Comment:
@stv0g The BatchConn interface has been used here which is implemented in the pion/transport.

Reply:
I mean, I dont find any code in pion/ice which calls `WriteBatch` or `ReadBatch`.
Maybe I am overseeing something, but having a batched UDPMux conn only helps if its also used by `ice.Conn`.

See: https://github.com/pion/ice/blob/d41eb5aae74782d5abb117efb52187f8caa06710/transport.go#L70

Reply:
Why ice need to call WriteBatch or ReadBatch? It has been wrapped in transport/udp.BatchConn

Reply:
@stv0g pion/transport uses batching internally. `WriteTo` would enqueue packets for batching [here](https://github.com/pion/transport/blob/master/udp/batchconn.go#L109)

Reply:
Thanks. Understood :)